### PR TITLE
ACSBP-3718 Update to use openapi 3.1

### PR DESCRIPTION
**Pull request by auramix**
These changes represent updating to use OpenApi 3.1, which will represent a major version bump for flask-rebar due to OpenApi's move away from semver in 3.1 and introduction of breaking changes. For a complete list of changes effecting our implementation of a v3 Generator, please see the attached ticket.
 JIRA Tickets | 
 --------------| 
 [ACSBP-3718](https://jira.autodesk.com/browse/ACSBP-3718)|

### CODE REVIEW

**Comment by gtmanfred** [comment]
Looks like we still need to validate the new swaggerui code here? https://github.com/plangrid/flask-rebar/tree/master/flask_rebar/swagger_ui

**Comment by auramix** [comment]
> Looks like we still need to validate the new swaggerui code here? https://github.com/plangrid/flask-rebar/tree/master/flask_rebar/swagger_ui

I could use a little context on that, as I'm not 100% clear on the function the swagger_ui directory serves. It appears to be home to static assets for generating a ui template that services can use to serve their swagger spec? I thought maybe you meant I should regenerate these static files, but after trying the create a blueprint with the provided function that didn't do what I expected.

**Comment by gtmanfred** [comment]
No, we just need to grab them from the dist directory in this release, and see if it can load correctly the swagger 3.1 artifacts 
https://github.com/swagger-api/swagger-ui/releases/tag/v5.0.0-alpha.13

**Comment by gtmanfred** [comment]
Here is a link to them in github https://github.com/swagger-api/swagger-ui/tree/v5.0.0-alpha.13/dist

**Comment by gtmanfred** [comment]
basically, the swagger-ui files vendored in flask-rebar right now haven't been updated in 5 years, we just need to update and vendor the new ones.

**Comment by gtmanfred** [comment]
I think we are still in the alpha release stages right?

File: README.rst [hunk 1]
```diff
@@ -111,6 +111,15 @@ Installation
...
 
 
+Replacing static swagger-ui files
+-------------
+
+If you'd like to replace swagger-ui's static files (`flask_rebar/swagger_ui/static`) with those of the latest release,
+run the following from the root of the project.
+
+.. code-block::
+    curl -L https://api.github.com/repos/swagger-api/swagger-ui/tarball | tar -xv --directory=flask_rebar/swagger_ui/static --strip-components=2 "*/dist/"
```
Comment (gtmanfred) [inline README.rst:121]: nice

**Review by nlawrence22 — APPROVED** [review]
This looks good to me, but I would definitely get a second opinion from one of the more seasoned flask developers on the team.

File: .flake8 [hunk 2]
```diff
@@ -3,3 +3,5 @@ max-line-length = 120
 exclude = .git/*, build/*, examples/*, venv/*, setup.py
 per-file-ignores =
     flask_rebar/__init__.py: F401 flask_rebar/*/__init__.py: F401
+select = B950
+extend-ignore = E501
```
Comment (airstandley) [inline .flake8:7]: I don't have a problem with us switching to B950 instead of E501, but this change should be made in a separate PR.

File: flask_rebar/swagger_generation/marshmallow_to_swagger.py [hunk 3]
```diff
@@ -295,11 +295,16 @@ def convert(self, obj, context):
                         )
                     )
 
-        if context.openapi_version == 3 and obj.allow_none:
-            jsonschema_obj["nullable"] = True
-
         return jsonschema_obj
 
+    # With OpenApi 3.1 nullable has been removed entirely
+    # and allowing 'none' means we return an array of allowed types that includes sw.null
```
Comment (airstandley) [inline flask_rebar/swagger_generation/marshmallow_to_swagger.py:301]: Thanks for the explanation!

**Review by airstandley — APPROVED** [review]
Looks good 👍
